File: src/main/java/nl/hu/adsd/dtmreserveringen/entity/ItemReservation.java
package nl.hu.adsd.dtmreserveringen.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDate;


@Getter
@Setter
@Entity
@ToString
@NoArgsConstructor
@Table(name = "item_reservation")
public class ItemReservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @NotNull
    private LocalDate reservationDate;
    @NotNull
    private int reservationPeriod;


    @JsonManagedReference(value = "Item -> ItemReservations")
    @ManyToOne
    @JoinColumn(name = "item_id")
    private Item item;

    @JsonBackReference(value = "Reservation -> ItemReservations")
    @ManyToOne
    @JoinColumn(name = "reservation_id")
    private Reservation reservation;
}

File: src/main/resources/static/classes/itemReservation.js
export class ItemReservation {

    constructor(reservationDate, reservationPeriod, itemId) {
        this.reservationDate = reservationDate || "";
        this.reservationPeriod = reservationPeriod || 0;
        this.itemId = itemId || 0;
    }

    setValuesFromDbJson(dbJson) {
        this.reservationDate = new Date(dbJson.reservationDate);
        this.reservationPeriod = dbJson.reservationPeriod;
        this.itemId = dbJson.item.id;
    }

    setValuesFromObject(data) {
        this.reservationDate = new Date(data.reservationDate);
        this.reservationPeriod = data.reservationPeriod;
        this.itemId = data.itemId;
    }
}